packages/git/src/browser/git-repository-provider.ts

```diff
@@ -43,7 +46,8 @@ export class GitRepositoryProvider {
         this.workspaceService.onWorkspaceChanged(event => {
             this.refresh();
         });
-        await this.refresh({ maxCount: 1 });
+        this._selectedRepository = await this.storageService.getData<Repository | undefined>(this.storageKey);
+        // await this.refresh({ maxCount: 1 });
```

Comment:
Why did we do that? If there are more than one repos it wouldn't be possible to set another one than the first of all repositories.

Reply:
it is an optimization for big projects with multiple nested repositories to report the root one fast and look up the rest later, please put it back.

> I have commented that out to get the restoring work, but I am not sure if I may do that.

How does it break restoring?